src/app/public/modules/repeater/repeater.component.ts

```diff
@@ -33,36 +47,92 @@ export class SkyRepeaterComponent implements AfterContentInit {
   @ContentChildren(SkyRepeaterItemComponent)
   public items: QueryList<SkyRepeaterItemComponent>;
 
+  private ngUnsubscribe = new Subject<void>();
+
   private _expandMode = 'none';
 
-  constructor(private repeaterService: SkyRepeaterService) {
-    this.repeaterService.itemCollapseStateChange.subscribe((item: SkyRepeaterItemComponent) => {
-      if (this.expandMode === 'single' && item.isExpanded) {
-        this.items.forEach((otherItem) => {
-          if (otherItem !== item && otherItem.isExpanded) {
-            otherItem.isExpanded = false;
-          }
-        });
-      }
-    });
+  constructor(
+    private repeaterService: SkyRepeaterService
+  ) {
+    this.repeaterService.itemCollapseStateChange
+      .takeUntil(this.ngUnsubscribe)
+      .subscribe((item: SkyRepeaterItemComponent) => {
+        if (this.expandMode === 'single' && item.isExpanded) {
+          this.items.forEach((otherItem) => {
+            if (otherItem !== item && otherItem.isExpanded) {
+              otherItem.isExpanded = false;
+            }
+          });
+        }
+      });
 
     this.updateForExpandMode();
   }
 
   public ngAfterContentInit() {
+    // Initialize each item's index (in case items are instantiated out of order).
+    this.items.forEach(item => item.initializeItem());
+    this.items.changes
+      .takeUntil(this.ngUnsubscribe)
+      .subscribe((change: QueryList<SkyRepeaterItemComponent>) => {
+        change
+          .filter(item => this.repeaterService.items.indexOf(item) < 0)
+          .forEach(item => item.initializeItem());
+      });
+
+    // If activeIndex has been set on init, call service to activate the appropriate item.
+    setTimeout(() => {
+      if (this.activeIndex || this.activeIndex === 0) {
+        this.repeaterService.activateItemByIndex(this.activeIndex);
+      }
+    });
+
+    // Watch for changes on the service's activeItemId and activate the appropriate item.
+    this.repeaterService.activeItemId
+      .takeUntil(this.ngUnsubscribe)
+      .distinctUntilChanged()
+      .subscribe((activeItemId) => {
+        // HACK: Not selecting the active item in a timeout causes an error.
+        // https://github.com/angular/angular/issues/6005
+        setTimeout(() => {
+          const activeItem = this.items.find(item => {
+            return item.itemId === activeItemId;
```

Comment:
This is more a question than a critique, but I thought we decided against using `itemId` and instead chose to rely on an index? It feels strange to be using both an index and an ID for the active item; I would go all-in on one. 

For the case where you need to alert the component of the active item changing, could you instead do an "equals" check against the object itself?

```
// The service returns the entire item object.
this.repeaterService.activeItemChange
  .subscribe((activeItem: SkyRepeaterItemComponent) = {
    const foundItem = this.items.find((item) => {
      return (item === activeItem);
    });
    // ...
  });
```

Reply:
Didn't think about comparing the whole item object. That seems to work great, thanks for the tip. Done.